### PR TITLE
Replace global consts with PluginInfo object

### LINKED ISSUE #599: Sharing variables and objects is a basic necessity
From https://github.com/leoloso/PoP/blob/f134be00ccf8e4b4ce788dbd4d801e5641834b3a/layers/GraphQLAPIForWP/plugins/graphql-api-for-wp/phpstan.neon.dist#L13

## Detailed description

# 🦕

Global constants `return \GRAPHQL_API_DIR;`
and static properties, functions `protected static $instance;`

# 👴

[A DI container](https://github.com/szepeviktor/injector-php-7) and [an immutable container](https://github.com/szepeviktor/small-project/blob/346d4c3008a2fc67f47df19a5b54c9546a27f135/plugin-name.php#L75-L86) are the object oriented solutions. Available since early PHP versions.

I let you take the wheel!

### PR DESCRIPTION
**Pull request by leoloso**
Fixes #599

### CODE REVIEW

File: layers/GraphQLAPIForWP/plugins/convert-case-directives/graphql-api-convert-case-directives.php [hunk 1]
```diff
@@ -23,19 +26,58 @@
...
+        });
+        return;
+    }
 
     // Load Composer’s autoloader
     require_once(__DIR__ . '/vendor/autoload.php');
 
-    (new \GraphQLAPI\ConvertCaseDirectives\GraphQLAPIExtension(__FILE__))->setup();
+    // Initialize the Plugin information
+    PluginInfo::init([
```
Comment (szepeviktor) [inline layers/GraphQLAPIForWP/plugins/convert-case-directives/graphql-api-convert-case-directives.php:76]: > PluginInfo

@leoloso What a name!

File: layers/GraphQLAPIForWP/plugins/convert-case-directives/graphql-api-convert-case-directives.php [hunk 2]
```diff
@@ -23,19 +26,58 @@
...
+        'version' => '0.7.13',
+        'dir' => dirname(__FILE__),
+        'url' => plugin_dir_url(__FILE__),
```
Comment (szepeviktor) [inline layers/GraphQLAPIForWP/plugins/convert-case-directives/graphql-api-convert-case-directives.php:79]: @leoloso Storing `plugin_dir_url()` out at any time prevents CDN - and other - plugins from working 😢

Comment (leoloso) [reply layers/GraphQLAPIForWP/plugins/convert-case-directives/graphql-api-convert-case-directives.php:79]: Good point! But that is just for internal use, to load the images for the documentation, displayed in the wp-admin, so it should work fine. (Also that's just for DEV, for PROD the images are excluded from the plugin, and retrieved from GitHub)

File: layers/GraphQLAPIForWP/plugins/graphql-api-for-wp/graphql-api.php [hunk 3]
```diff
@@ -15,40 +15,36 @@
...
-use GraphQLAPI\GraphQLAPI\Plugin;
-
 /**
  * Make sure this plugin is not duplicated.
  * For instance, if zip file already exists in Downloads folder, then
  * the newly downloaded file will be renamed (eg: graphql-api(2).zip)
  * and the plugin will exist twice, as graphql-api/... and graphql-api2/...
  */
-if (defined('GRAPHQL_API_VERSION')) {
+if (class_exists('\GraphQLAPI\GraphQLAPI\PluginInfo')) {
```
Comment (szepeviktor) [inline layers/GraphQLAPIForWP/plugins/graphql-api-for-wp/graphql-api.php:32]: @leoloso PHP 5 has `::class` - so `class_exists(PluginInfo::class)`

Comment (leoloso) [reply layers/GraphQLAPIForWP/plugins/graphql-api-for-wp/graphql-api.php:32]: But if class PluginInfo doesn't exist, then `PluginInfo::class` will throw an error! That's why I need to pass it as a string, so it never fails

Comment (szepeviktor) [reply layers/GraphQLAPIForWP/plugins/graphql-api-for-wp/graphql-api.php:32]: Are you sure? https://3v4l.org/tdg9H

Comment (leoloso) [reply layers/GraphQLAPIForWP/plugins/graphql-api-for-wp/graphql-api.php:32]: Interesting, it seems you're right! OK, I'll change that in my next PR

**Pull request merged**